pkg/option/config.go

```diff
@@ -3082,8 +3112,14 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 	c.TCFilterPriority = uint16(tcFilterPrio)
 
 	c.Tunnel = vp.GetString(TunnelName)
+	c.RoutingMode = vp.GetString(RoutingMode)
+	c.TunnelProtocol = vp.GetString(TunnelProtocol)
 	c.TunnelPort = vp.GetInt(TunnelPortName)
 
+	if c.Tunnel != "" && c.RoutingMode != defaults.RoutingMode {
+		log.Fatalf("Option --%s cannot be used in combination with --%s", RoutingMode, TunnelName)
```

Comment:
Nit: Why fatal here when most* of the function is returning errors?

*: The other inconsistency is on L3310 and L3103. All others return errors.

Reply:
@christarazi Hm, I'm missing something here. There are 19 other fatals in this same function. The function also doesn't return anything anyway.

Reply:
Oh I think I was looking at the wrong function, nevermind.